ios/engine/KMEI/KeymanEngine/Classes/Migrations.swift

```diff
@@ -490,4 +501,180 @@ public enum Migrations {
       }
     }
   }
+
+  static func migrateCloudResourcesToKMPFormat() throws {
+    let userDefaults = Storage.active.userDefaults
+
+    if var userKeyboards = userDefaults.userKeyboards {
+      userKeyboards = migrateToKMPFormat(userKeyboards)
+      userDefaults.userKeyboards = userKeyboards
+    }
+
+    if var userLexicalModels = userDefaults.userLexicalModels {
+      userLexicalModels = migrateToKMPFormat(userLexicalModels)
+      userDefaults.userLexicalModels = userLexicalModels
+    }
+  }
+
+  static func migrateToKMPFormat<Resource, Package: TypedKeymanPackage<Resource>>(
+        _ resources: [Resource]) -> [Resource]
+        where Resource: KMPInitializableLanguageResource,
+              Resource == Resource.Metadata.LanguageResourceType {
+    // Step 1 - drop version numbers from all filenames.  KMP installations don't include them, and
+    // the version numbered filenames will actually interfere with migration.
+    resources.forEach { resource in
+      let srcLocation = Storage.active.legacyResourceURL(for: resource)!
+      let dstLocation = Storage.active.resourceURL(for: resource)!
+
+      do {
+        if FileManager.default.fileExists(atPath: srcLocation.path) {
+          try FileManager.default.moveItem(at: srcLocation, to: dstLocation)
+        }
+      } catch {
+        log.error("Could not remove version number from filename for resource \(resource.packageID ?? "<no package>").\(resource.id)")
+        log.error("Source: \(srcLocation)")
+      }
+    }
+
+    // Step 2 - analyze all locally-cached KMPs for possible resource sources.
+    var allLocalPackages: [(KeymanPackage, URL)] = []
+    do {
+      let cachedKMPsDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]
+      let files = try FileManager.default.contentsOfDirectory(atPath: cachedKMPsDirectory.path)
+      let kmpFiles = files.filter { $0.suffix(4) == ".kmp" }
```

Comment:
Whenever we're checking file extensions, should we also account for ".KMP" and ".MODEL.KMP"?